runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -136,22 +136,32 @@ private MetricQueryResults populateMetricQueryResults(
         ImmutableList.builder();
     ImmutableList.Builder<MetricResult<GaugeResult>> gaugeResults = ImmutableList.builder();
     for (MetricKey metricKey : metricHashKeys) {
-      String metricName = metricKey.metricName().name();
-      if (metricName.endsWith("[MIN]") || metricName.endsWith("[MAX]")
-          || metricName.endsWith("[MEAN]") || metricName.endsWith("[COUNT]")) {
-        // Skip distribution metrics, as these are not yet properly supported.
-        LOG.warn("Distribution metrics are not yet supported. You can see them in the Dataflow"
-            + "User Interface");
+      if (!MetricFiltering.matches(filter, metricKey)) {
+        // Skip unmatched metrics early.
         continue;
       }
-      String namespace = metricKey.metricName().namespace();
-      String step = metricKey.stepName();
-      Long committed = ((Number) committedByName.get(metricKey).getScalar()).longValue();
-      Long attempted = ((Number) tentativeByName.get(metricKey).getScalar()).longValue();
-      if (MetricFiltering.matches(filter, metricKey)) {
-        counterResults.add(DataflowMetricResult.create(
-            MetricName.named(namespace, metricName),
-            step, committed, attempted));
+
+      // This code has is not robust to evolutions in the types of metrics that can be returned, so
```

Comment:
Might want to be more specific about the responsibility here. I don't know exactly what "this code" refers to. Is some contract by which `metricKey` is required to be in `committedByName` and `tentativeByName`? In other words, is the bug in building those maps or in accessing them.

If the latter, just checking for `null` seems as quick a fix.